include/ISceneNode.h

```diff
@@ -624,6 +623,9 @@ namespace scene
 		//! List of all children of this node
 		std::list<ISceneNode*> Children;
 
+		//! Iterator pointing to this node in the parent's child list.
+		std::optional<ISceneNodeList::iterator> ThisIterator;
```

Comment:
Maybe put it right after `Parent` to show that they're essentially related.

Reply:
I moved Parent right after this, so this is now between Parent and Children, the two fields its related to (the parent's children and this node's parent). I think that's a decent compromise.